### PR TITLE
Add Visual Storybook integration example

### PR DESCRIPTION
**Pull request by omacranger**
Adds an example setting up visual integration with a basic CRA & Storybook app

- Project was created using `npx create-react-app DIR_NAME --template typescript`
- Added Storybook using `npx storybook@latest init`
- Followed [README instructions](https://www.npmjs.com/package/@saucelabs/visual-storybook) for `@saucelabs/visual-storybook` package

To run locally, follow step number 5 on the [getting started readme](https://www.npmjs.com/package/@saucelabs/visual-storybook#getting-started).

### CODE REVIEW

File: visual-storybook/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- Start Storybook locally in a separate terminal window and leave running in the background
+
+```sh { name=start-storybook }
+npm run storybook
+```
+
+- Configure with your Sauce credentials from https://app.saucelabs.com/user-settings and run
+
+```sh { name=set-credentials }
+export SAUCE_USERNAME=__YOUR_SAUCE_USER_NAME__
```
[the remark this answers is not in the document]
Comment (omacranger) [reply visual-storybook/README.md:25]: Yep good call.

File: visual-storybook/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export SAUCE_ACCESS_KEY=__YOUR_SAUCE_ACCESS_KEY__
+# You can change the region to upload to via the SAUCE_REGION env variable. See the 
+# @saucelabs/visual-storybook README for more details on how to customize your run. 
+```
+
+- Run the test
+
+```sh { name=npm-run }
+npm run test-storybook
```
[the remark this answers is not in the document]
Comment (omacranger) [reply visual-storybook/README.md:34]: Yeah this is a one time thing. I can add it to the docs here with that note though.

File: visual-storybook/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Getting started with Storybook
```
[the remark this answers is not in the document]
Comment (omacranger) [reply visual-storybook/README.md]: Done.

File: visual-storybook/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (omacranger) [reply visual-storybook/README.md]: This is also done, but going to leave this thread unresolved until I can publish the new package and adopt the minimum version in the package.json here.

File: visual-storybook/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+```sh { name=npm-run }
+npm run test-storybook
+```
+
+- Go to https://app.saucelabs.com/visual/builds and select your latest Storybook run
+- Accept all diffs, so they become new baselines.
+- Make some changes in one of the Storybook Stories
+
+For an example, let's make the following changes in `visual-storybook/src/stories/Button.stories.ts`:
```
[the remark this answers is not in the document]
Comment (omacranger) [reply visual-storybook/README.md]: Added & updated readme

**Comment by paweltomaszewskisaucelabs** [comment]
@omacranger I found one more issue. It generates two snapshots for "Logged In" and "Logged Out" and it can cause confusion to users. Can you fix it too?

Especially that two snapshots with the same name in the build are referring to the same baseline. (I think we should handle somehow creating snapshots with the same name in the backend)

**Comment by omacranger** [comment]
@paweltomaszewskisaucelabs 

This is fixed in my MR awaiting review for the visual-storybook package as well. It's now including suite & story name when creating titles.

**Review by paweltomaszewskisaucelabs — APPROVED** [review]
Please make sure that USERNAME env and suite & story issues are fixed before merging

**Pull request merged**